Gordon360/Services/RecIM/SportService.cs

```diff
@@ -78,39 +78,42 @@ public async Task<SportViewModel> UpdateSportAsync(int sportID, SportPatchViewMo
             sport.Rules = updatedSport.Rules ?? sport.Rules;
             
             // UNTESTED FEATURE
-            if (updatedSport.Logo != null)
+            if (updatedSport.IsLogoUpdate)
             {
-                // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
-                var (extension, format, data) = ImageUtils.GetImageFormat(updatedSport.Logo);
-
-                string? imagePath = null;
-                // If old image exists, overwrite it with new image at same path
-                if (sport.Logo != null)
+                if (updatedSport.Logo != null)
                 {
-                    imagePath = GetImagePath(Path.GetFileName(sport.Logo));
+                    // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
+                    var (extension, format, data) = ImageUtils.GetImageFormat(updatedSport.Logo);
+
+                    string? imagePath = null;
+                    // If old image exists, overwrite it with new image at same path
+                    if (sport.Logo != null)
+                    {
+                        imagePath = GetImagePath(Path.GetFileName(sport.Logo));
+                    }
+                    // Otherwise, upload new image and save url to db
+                    else
+                    {
+                        // Use a unique alphanumeric GUID string as the file name
+                        var filename = $"{Guid.NewGuid().ToString("N")}.{extension}";
+                        imagePath = GetImagePath(filename);
+                        var url = GetImageURL(filename);
+                        sport.Logo = url;
+                    }
+
+                    ImageUtils.UploadImage(imagePath, data, format);
```

Comment:
Seems like a good amount of repeated code as earlier. Perhaps this could be factored out instead of repeated? Generic Recim service?

Reply:
Huh interesting, also I have no idea what was going through my mind when thinking about having an `IEnumerable`, I think my mind was in 5 separate places and I was thinking about how we would implement the image update in its own `RecIMSerivces` and lost track of where my ideas were.